### PR TITLE
Converting xml tests to mocha.

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

#2870

### Proposed Changes
Converts remaining jsunit tests in xml_test.js to mocha.
Also, refactors code in mocha/xml_test.js
- Replaces uses of try/finally with setup/teardown where appropriate
- Refactors some shared code
- Renames test suites to be clearer what method is being tested
- Replaces monkey patching with use of `sinon.stub`

### Reason for Changes
Explained in linked issue.

### CODE REVIEW

File: tests/mocha/xml_test.js [hunk 1]
```diff
@@ -653,53 +824,63 @@ suite('XML', function() {
...
+        type: type,
+        name: name,
+        getId: function() {
+          return varId;
+        }
+      };
+
+      var generatedXml = Blockly.Xml.domToText(
+          Blockly.Variables.generateVariableFieldDom(mockVariableModel));
+      var goldenXml =
```
Comment (alschmiedt) [inline tests/mocha/xml_test.js:877]: expectedXml to keep consistent with above.

File: tests/mocha/xml_test.js [hunk 2]
```diff
@@ -369,143 +415,153 @@ suite('XML', function() {
...
+
+      var resultDom = Blockly.Xml.variablesToDom(this.workspace.getAllVariables());
+      assertEquals(2, resultDom.children.length);
+      assertVariableDom(resultDom.children[0], null, 'id1',
+          'name1');
+      assertVariableDom(resultDom.children[1], 'type2', 'id2',
+          'name2');
+    });
+    test('No variables', function() {
+      // TODO(kozbial) does this test make sense? copied from jsunit/xml_test
```
Comment (alschmiedt) [inline tests/mocha/xml_test.js:466]: Remove TODO

File: tests/mocha/xml_test.js [hunk 3]
```diff
@@ -640,7 +693,125 @@ suite('XML', function() {
...
+      var dom = Blockly.Xml.textToDom(
+          '<xml xmlns="https://developers.google.com/blockly/xml">' +
+          '  <block type="test_block" inline="true" x="21" y="23">' +
+          '  </block>' +
+          '</xml>');
+      Blockly.Xml.appendDomToWorkspace(dom, this.workspace);
+      assertEquals('Block count', 1, this.workspace.getAllBlocks(false).length);
+      var newBlockIds = Blockly.Xml.appendDomToWorkspace(dom, this.workspace);
+      assertEquals('Block count', 2, this.workspace.getAllBlocks(false).length);
+      assertEquals('Number of new block ids',1,newBlockIds.length);
```
Comment (alschmiedt) [inline tests/mocha/xml_test.js:811]: Addd spaces between commas

File: tests/mocha/xml_test.js (hunk 2 above)
Comment (moniika) [reply tests/mocha/xml_test.js:466]: Updated test case and removed TODO

File: tests/mocha/xml_test.js (hunk 3 above)
Comment (moniika) [reply tests/mocha/xml_test.js:811]: Done.

File: tests/mocha/xml_test.js (hunk 1 above)
Comment (moniika) [reply tests/mocha/xml_test.js]: Done.

**Pull request merged**